did_peer/src/numalgos/numalgo2/resolve/helpers.rs

```diff
@@ -0,0 +1,413 @@
+use base64::{engine::general_purpose::STANDARD_NO_PAD, Engine};
+use did_doc::schema::{did_doc::DidDocumentBuilder, service::Service, types::uri::Uri, utils::OneOrList};
+use did_doc_sov::extra_fields::{aip1::ExtraFieldsAIP1, didcommv2::ExtraFieldsDidCommV2, ExtraFieldsSov};
+use did_parser::Did;
+
+use crate::{
+    error::DidPeerError,
+    key::{get_verification_methods_by_key, Key},
+    numalgos::numalgo2::{purpose::ElementPurpose, service_abbreviated::ServiceAbbreviated},
+    peer_did_resolver::options::PublicKeyEncoding,
+};
+
+pub fn process_elements(
+    mut did_doc_builder: DidDocumentBuilder<ExtraFieldsSov>,
+    did: &Did,
+    public_key_encoding: &PublicKeyEncoding,
+) -> Result<DidDocumentBuilder<ExtraFieldsSov>, DidPeerError> {
+    let numalgoless_id = did.id().chars().skip(1).collect::<String>();
+    let mut service_index: usize = 0;
+
+    // Skipping one here because the first element is empty string
+    for element in numalgoless_id.split('.').skip(1) {
```

Comment:
How about something like this? Avoids allocation, skips the first char like before *and* the first `.` that I assume is causing the empty string for the first element.

```suggestion
    let numalgoless_id = &did.id()[2..];
    let mut service_index: usize = 0;
    
    for element in numalgoless_id.split('.') {
```